flask_mongorest/__init__.py

```diff
@@ -2,16 +2,64 @@
 from flask_mongorest.methods import Create, BulkUpdate, List
 
 
+class _DelayedApp(object):
+
+    '''
+    Stores URL rules for later merging with application URL map.
+
+    '''
+
+    def __init__(self):
+        self.url_rules = []
+
+    def add_url_rule(self, *args, **kwargs):
+        self.url_rules.append((args, kwargs))
+
+
 class MongoRest(object):
-    def __init__(self, app, **kwargs):
-        self.app = app
+
+    def __init__(self, app=None, **kwargs):
+        '''
+        Takes optional Flask application instance. If supplied, `init_app` will be
+        called to update application url map.
+
+        '''
+
         self.url_prefix = kwargs.pop('url_prefix', '')
-        app.register_blueprint(Blueprint(self.url_prefix, __name__, template_folder='templates'))
+        self.template_folder = kwargs.pop('template_folder', 'templates')
```

Comment:
Is there a reason why we pop a `template_folder` from `kwargs` here instead of specifying `def __init__(self, app=None, template_folder='templates', **kwargs)` above?